eva-accession-import/src/main/java/uk/ac/ebi/eva/accession/dbsnp/processors/ContigReplacerProcessor.java

```diff
@@ -31,33 +34,39 @@
 
     private String assemblyAccession;
 
+    private Set<String> processedContigs;
+
     public ContigReplacerProcessor(ContigMapping contigMapping, String assemblyAccession) {
         this.contigMapping = contigMapping;
         this.assemblyAccession = assemblyAccession;
+        this.processedContigs = new HashSet<>();
     }
 
     @Override
     public SubSnpNoHgvs process(SubSnpNoHgvs subSnpNoHgvs) throws Exception {
-        ContigSynonyms contigSynonyms = contigMapping.getContigSynonyms(subSnpNoHgvs.getContigName());
+        String contigName = subSnpNoHgvs.getContigName();
+        ContigSynonyms contigSynonyms = contigMapping.getContigSynonyms(contigName);
         ContigSynonyms chromosomeSynonyms = contigMapping.getContigSynonyms(subSnpNoHgvs.getChromosome());
 
         boolean chromosomePresentInAssemblyReport = chromosomeSynonyms != null;
         boolean contigPresentInAssemblyReport = contigSynonyms != null;
 
         if (!contigPresentInAssemblyReport && !chromosomePresentInAssemblyReport) {
             throw new IllegalStateException(
-                    "Neither contig '" + subSnpNoHgvs.getContigName() + "' nor chromosome '"
+                    "Neither contig '" + contigName + "' nor chromosome '"
                             + subSnpNoHgvs.getChromosome()
                             + "' were found in the assembly report! Is the assembly accession '"
                             + assemblyAccession + "' correct?");
         }
 
-        if (chromosomePresentInAssemblyReport
+        if (!processedContigs.contains(contigName)
+                && chromosomePresentInAssemblyReport
                 && contigPresentInAssemblyReport
```

Comment:
really pedantic, but putting the check for contains after both boolean checks will save lots of strings hash creation.

Reply:
+1